src/biokbase/narrative/tests/test_appmanager.py

```diff
@@ -1107,118 +1113,100 @@ def test_transform_input_bad(self):
-                self.assertEqual(message_name, expected_messages[i])
-            if expected_keys is not None:
-                for key in expected_keys[i]:
-                    self.assertIn(key, message_dict)
-            if expected_values is not None:
-                for key in expected_values[i]:
-                    self.assertEqual(message_dict.get(key), expected_values[i][key])
+    def _transform_comm_messages(self, comm_mock):
+        transformed_call_args_list = []
+        for call in comm_mock.call_args_list:
+            [msg_type, content] = call[0]
+            transformed_call_args_list.append(
+                {"msg_type": msg_type, "content": content}
+            )
+        return transformed_call_args_list
 
     def _verify_comm_error(self, comm_mock, cell_id=None, run_id=None) -> None:
         """
-        a wrapper around self._verify_comm_mock that just looks for the error message, and
-        ONLY the error message, to have been sent, so we can make these tests a little more
-        DRY.
-
-        This is really for app running cases that are expected to fail before trying to start
-        the app with EE2. Other cases should run _verify_comm_mock directly.
+        Generates standard error messages and tests it against stored comm messages.
         """
-        expected_keys = [
-            [
-                "event",
-                "event_at",
-                "error_message",
-                "error_code",
-                "error_source",
-                "error_stacktrace",
-                "error_type",
-            ]
-        ]
-        expected_values = [
-            {
-                "run_id": run_id,
-                "cell_id": cell_id,
+        transformed_call_args_list = self._transform_comm_messages(comm_mock)
+        expected_message = {
+            "msg_type": RUN_STATUS,
+            "content": {
                 "event": "error",
+                "event_at": get_timestamp(),
                 "error_code": -1,
                 "error_source": "appmanager",
-            }
-        ]
-        self._verify_comm_mock(
-            comm_mock,
-            1,
-            expected_messages=[RUN_STATUS],
-            expected_keys=expected_keys,
-            expected_values=expected_values,
-        )
+                "error_type": "ValueError",
+            },
+        }
+        if run_id:
+            expected_message["content"]["run_id"] = run_id
+        if cell_id:
+            expected_message["content"]["cell_id"] = cell_id
 
-    def _verify_comm_success(
-        self, comm_mock, is_batch, num_jobs=1, cell_id=None, run_id=None
-    ) -> None:
-        expected_messages = [RUN_STATUS]
-        expected_keys = [
-            ["event", "event_at", "cell_id", "run_id"],
+        for key in ["error_message", "error_stacktrace"]:
+            self.assertTrue(key in transformed_call_args_list[0]["content"])
+            del transformed_call_args_list[0]["content"][key]
```

Comment:
these are different for each error and difficult to autogenerate, so just ensure these keys are present and then ignore them